record-and-playback/presentation/scripts/publish/presentation.rb

```diff
@@ -849,7 +849,18 @@ def processChatMessages
 						chat_sender = node.xpath(".//sender")[0].text()
 						chat_message =  BigBlueButton::Events.linkify(node.xpath(".//message")[0].text())
 						chat_start = ( translateTimestamp(chat_timestamp) / 1000).to_i
-						$xml.chattimeline(:in => chat_start, :direction => :down,  :name => chat_sender, :message => chat_message, :target => :chat )
+						#this chat_end argument will carry the time to remove a message if a clear was issued
+						chat_end = ( translateTimestamp(re[:stop_timestamp]) / 1000).to_i
+						#unless there is a clear event (after the messge was sent) it will be there until the end of the recording 
+						unless $clear_chat_events.last[:timestamp].to_i <= node[:timestamp].to_i
+							$clear_chat_events.reverse_each do |clear|
+								#but if there is a clear after the message, then the first clear after it will be the end time of this message
+								if (clear[:timestamp].to_i >= node[:timestamp].to_i)
+									chat_end = ( translateTimestamp(clear[:timestamp]) / 1000).to_i
+								end
+							end
+						end
+						$xml.chattimeline(:in => chat_start, :out => chat_end, :direction => :down,  :name => chat_sender, :message => chat_message, :target => :chat )
```

Comment:
I think it can be simplified. Imagine this:
- process clear_chat_events so you have only the timestamps, not the xml nodes
- include $meeting_end at the end of clear_chat_events
- then you can set chat_end as the first element of the sorted clear_chat_events selected to have only the elements with timestamp equal or higher than the chat message timestamp
It would essentially reduce this code to one line very simple to understand.